### PR TITLE
CI: support Rails v7.1

### PR DESCRIPTION
**Pull request by sgerrand**
💁 These changes add Rails version 7.1.x to the test matrix. I upgraded the `nio4r` gem as part of these changes because the previous version threw compilation errors on the initial installation attempt.

### CODE REVIEW

File: Gemfile.lock [hunk 1]
```diff
@@ -139,7 +139,7 @@ GEM
     minitest (5.16.3)
     multi_json (1.15.0)
     multi_xml (0.6.0)
-    nio4r (2.5.8)
+    nio4r (2.7.3)
```
Comment (sgerrand) [inline Gemfile.lock:142]: Updated to resolve the following.

> 2 warnings generated.
> 2 warnings generated.
> 89 warnings generated.
>
> make failed, exit code 2

File: gemfiles/rails_6.gemfile.lock [hunk 2]
```diff
@@ -261,6 +261,7 @@ GEM
 
 PLATFORMS
   arm64-darwin-21
+  arm64-darwin-22
```
Comment (sgerrand) [inline gemfiles/rails_6.gemfile.lock:264]: Added this because that's the platform I installed the gems on.

File: gemfiles/rails_7.gemfile.lock [hunk 3]
```diff
@@ -270,6 +270,7 @@ GEM
 
 PLATFORMS
   arm64-darwin-21
+  arm64-darwin-22
```
Comment (sgerrand) [inline gemfiles/rails_7.gemfile.lock:273]: Added this because that's the platform I installed the gems on.

File: gemfiles/rails_7_1.gemfile.lock [hunk 4]
```diff
@@ -0,0 +1,327 @@
...
+    sprockets (4.2.1)
+      concurrent-ruby (~> 1.0)
+      rack (>= 2.2.4, < 4)
+    sprockets-rails (3.4.2)
+      actionpack (>= 5.2)
+      activesupport (>= 5.2)
+      sprockets (>= 3.0.0)
+    sqlite3 (1.7.0-arm64-darwin)
+    sqlite3 (1.7.0-x86_64-darwin)
+    sqlite3 (1.7.0-x86_64-linux)
```
Comment (sgerrand) [inline gemfiles/rails_7_1.gemfile.lock:282]: [ActiveRecord's SQLite3 adapter _still_ forces the gem version to `"~> 1.4"`](https://github.com/rails/rails/blob/v7.1.3.2/activerecord/lib/active_record/connection_adapters/sqlite3_adapter.rb#L14-L15), added this for safety. 😞

**Review by sgerrand — COMMENTED** [review]
📝

File: Gemfile.lock (hunk 1 above)
Comment (sgerrand) [reply Gemfile.lock:142]: See 6c43c7d1ef3ebc825b27de8e844a6fd9dd76fddc for all the gory details.

**Comment by igorkasyanchuk** [comment]
@sgerrand thanks for working on this. Please fix the specs too.

**Comment by sgerrand** [comment]
The errors in the GitHub Action workflow stem from variations on this setup error:

> You are trying to install in deployment mode after changing
your Gemfile. Run `bundle install` elsewhere and add the
updated gemfiles/rails_7_1.gemfile.lock to version control.
>
> If this is a development machine, remove the
/home/runner/work/any_login/any_login/gemfiles/rails_7_1.gemfile freeze
by running `bundle config unset deployment`.
>
> The lockfile does not have all gems needed for the current platform
> Error: The process '/opt/hostedtoolcache/Ruby/3.0.7/x64/bin/bundle' failed with exit code 16

I'm working on fixing those now.

**Comment by igorkasyanchuk** [comment]
btw you can delete Rails 6 from CI, I see that it's already has reached EOL

**Comment by sgerrand** [comment]
> btw you can delete Rails 6 from CI, I see that it's already has reached EOL

I might do that in a separate PR, if you don't mind.

**Comment by sgerrand** [comment]
I've rebased this branch. These changes should be good to review again.

**Comment by igorkasyanchuk** [comment]
Merged, thanks!

**Comment by igorkasyanchuk** [comment]
@sgerrand I have a question, do you know any company looking for a Ruby developer? I'll be available for work soon and want to use my connections. If yes - great, if not - also thanks for just thinking about it :)

**Comment by sgerrand** [comment]
Not at the moment but I'll be sure to mention your name if I happen to hear of anything.

**Pull request merged**